airbyte-db/lib/src/main/java/io/airbyte/db/factory/DataSourceFactory.java

```diff
@@ -147,6 +149,21 @@ public static DataSource createPostgres(final String username,
         .build();
   }
 
+  /**
+   * Utility method that attempts to close the provided {@link DataSource} if it implements
+   * {@link Closeable}.
+   *
+   * @param dataSource The {@link DataSource} to close.
+   * @throws IOException if unable to close the data source.
+   */
+  public static void close(final DataSource dataSource) throws IOException {
+    if (dataSource != null) {
+      if (dataSource instanceof Closeable closeable) {
```

Comment:
Probably safer to use `AutoCloseable` because 1) `Closeable` extends from `AutoCloseable`, and 2) the common `BasicDataSource` we use in lots of places is an `AutoCloseable` but not a `Closeable`.

```suggestion
      if (dataSource instanceof AutoCloseable closeable) {
```